src/GslCore/ResolveExtPart.fs

```diff
@@ -2,208 +2,122 @@
-                     description = rabit.Name; 
-                     sliceType = REGULAR; 
-                     amplified = false;
-                     template = Some dna; // not amplifying from this
-                     dnaSource =
-                        match ppp.pr.TryGetOne("dnasrc") with
-                        | Some(d) -> d
-                        | None -> pid;
-                     pragmas = ppp.pr;
-                     breed = B_X; // will be replaced at final submission
-                     materializedFrom = Some(ppp);
-                     annotations = []} // FIXME: need to generate annotations based on Rabit metadata
-                else
-                    // Otherwise, they are taking a hutch part and doing something to it,
-                    // so the hutch is just another DNA source and they are effectively
-                    // building a new rabit
+    // Have part from the hutch.  We might just use it verbatim or we might be
+    // some modifications to it to make a new part
+    let part = fetchPart pid |> returnOrFail
 
-                    // Start off assuming it's the full DNA slice
-                    let startSlice =
-                       {left = {x = 1<OneOffset>; relTo = FivePrime};
-                        lApprox = false; 
-                        rApprox = false;
-                        right = {x = -1<OneOffset>; relTo = ThreePrime}}
+    // Check for slice modifications.  We can't handle any other type of mod at this point, so
+    // ensure there are none.
+    if partId.mods |> List.exists (fun m -> match m with | SLICE(_)-> false | _ -> true) then
+        failwithf "ERROR:  could not process mods for rabit %s %A\n" partId.id partId.mods
 
-                    // Apply the slice(s) to get a final coordinate range
-                    let finalSlice = applySlices verbose partId.mods startSlice 
-
-                    // Find the left and right hand ends of the slice
-                    let x, y =
-                        getBoundsFromSlice finalSlice dna.Length (Library(partId.id))
-                        |> returnOrFail
-                
-                    let finalDNA =
-                        dna.[(x/1<OneOffset>)-1..(y/1<OneOffset>)-1]
-                        |> DnaOps.revCompIf (not ppp.fwd)
+    // Look for simple case.  If we are just using the part  unadulterated, then
+    // we specify things differently, referring to the external id.
+    if partId.mods.Length = 0 then
+        let dna = part.dna |> DnaOps.revCompIf (not ppp.fwd)
 
-                    let name1 =
-                        if partId.mods.Length = 0 then rabit.Name
-                        else (rabit.Name + (printSlice finalSlice))
-                    let name2 = if ppp.fwd then name1 else "!" + name1
+        {id = None;
+         extId = Some pid; 
+         sliceName = sliceName;
+         uri = uri; // TODO: use the URI of part here instead?
+         dna = dna; 
+         sourceChr = "library";

```

Comment:
I don't think this is necessary for this round, but long term can imagine part providers being able to fill this out more richly - e.g. source chromosome sourceChr could be more detailed.  I wouldn't break the world for round 1 though unless you can think of a slightly more generalized type signature for the returned part.